src/core/server/saved_objects/import/create_saved_objects.ts

```diff
@@ -125,6 +125,24 @@ export const createSavedObjects = async <T>({
               name: 'dataSource',
             });
           }
+
+          if (visStateObject.type === 'metrics') {
+            const oldDataSourceId = visStateObject.params.data_source_id;
+            object.references = object.references.filter((reference) => {
+              return reference.id !== oldDataSourceId && reference.type === 'data-source';
+            });
+
+            visStateObject.params.data_source_id = dataSourceId;
+
+            object.references.push({
+              id: dataSourceId,
+              name: 'dataSource',
+              type: 'data-source',
+            });
+
+            // @ts-expect-error
+            object.attributes.visState = JSON.stringify(visStateObject);
+          }
```

Comment:
Hi Huy. Looks like the code here is very similar to the check_conflict except check_conflict makes sure datasourceId exists. Can we combine these two and adds  it in the util?

Reply:
Refactored